src/System.Management.Automation/engine/debugger/debugger.cs

```diff
@@ -1313,12 +1321,11 @@ private void UpdateBreakpoints(FunctionContext functionContext)
                     return;
                 }
 
-                foreach ((int breakpointId, LineBreakpoint item) in _pendingBreakpoints)
+                if (_pendingBreakpoints.TryGetValue(functionContext._file, out var dictionary))
                 {
-                    if (item.IsScriptBreakpoint && item.Script.Equals(functionContext._file, StringComparison.OrdinalIgnoreCase))
+                    if (dictionary.Count > 0)
```

Comment:
I am not sure why `IsScriptBreakpoint` was checked here, but it was not re-checked anywhere else. `SetPendingBreakpoints` below is called **without** the list of breakpoints to set, and internally it only checks the filepath from the function context. So I skipped the check to avoid looping in case there are thousands of breakpoints in one file.